### ISSUE TITLE
Compiler segfault from `continue` inside runtime conditional inside `for` inside `inline for`

### ISSUE THREAD

**Issue by ghost** [ziglang/zig#4484]
```zig
pub fn main() void {
    var cond = true;

    inline for ([_]u8 { 0, 1, 2 }) |_| {
        for ([_]u8 { 3, 4, 5 }) |_| {
            if (!cond) {
                continue;
            }
            return;
        }
    }
}
```

```
Segmentation fault: 11
```

No crash if I delete the `return`. Same if I replace the loop body with `if (cond) return;`

I ran into this in my game earlier, with similar code, but instead of segfaulting the compiler was complaining (erroneously) that I was using a runtime condition to continue from a comptime loop. I tried to recreate that in my game now and the compiler crashed instead.

**Comment by daurnimator** [comment]
Dupe of #2727 ?

**Comment by Vexu** [comment]
Fixed by 4618c41fa6ca70f06c7e65762d2f38d57b00818c
